public/video-ui/src/pages/Search/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import VideoItem from '../../components/VideoItem';
+import { frontPageSize } from '../../constants/frontPageSize';
+import { getStore } from '../../util/storeAccessor';
+
+declare global {
+  interface Window { presenceClient?: any; }
+}
+
+type Video = {
+  id: string
+}
+
+type VideosProps = {
+  videos: Video[], 
+  total: number, 
+  videoActions: {
+    getVideos: (searchTerm: string, limit: number) => null;
+  },
+  searchTerm: string,
+  limit: number
+}
+
+type PresenceConfig = {
+  domain: string,
+  email: string,
+  firstName: string,
+  lastName: string
+}
+
+type Presence = {
+  clientId: {
+    connId: string,
+    person: {
+      browserId: string,
+      email: string,
+      firstName: string,
+      googleId: string,
+      lastName: string
+    },
+    lastAction: string, // ISO 8601 timestamp
+    location: string // Always seems to be "document"
+  },
+}
+
+type VideoPresences = {
+  mediaId: string,
+  presences: Presence[]
+}
+
+type PresenceData = {
+  subscriptionId?: string,
+  currentState?: Presence[],
+  subscribedTo: PresenceData[],
+  data?: PresenceData
+}
+
+type PresenceClient = {
+  on: (string: string, f: (data?: PresenceData ) => void) => void;
+  subscribe: (mediaIds: string[]) => void;
+  startConnection: () => void;
+}
+
+const MoreLink = ({ onClick }: {onClick: () => void} ) => {
+  return (<div>
+    <button className="btn video__load_more" onClick={onClick}>
+      Load More
+    </button>
+  </div>)
+}
+
+const Videos = ({videos, total, videoActions, searchTerm, limit}: VideosProps) => {
+  const [mediaIds, setMediaIds] = useState<string[]>([]);
+  const [videoPresences, setVideoPresences] = useState<VideoPresences[]>([]);
+  const [client, setClient] = useState<PresenceClient>(null);
+  const [presencesQueue, setPresencesQueue] = useState<PresenceData>(null);
+
+  const showMore = () => {
+    videoActions.getVideos(
+      searchTerm,
+      limit + frontPageSize
+    );
+  };
+
+  const startPresence = ({ domain, firstName, lastName, email }: PresenceConfig) => {
+    const endpoint = `wss://${domain}/socket`;
+    if (!window.presenceClient){
+      console.error("Failed to connect to Presence as client was not available in window.")
+      return;
+    }
+
+    const presenceClient = window.presenceClient(endpoint, {
+      firstName,
+      lastName,
+      email
+    }) as PresenceClient
+    presenceClient.startConnection();
+    presenceClient.on('visitor-list-subscribe', visitorData => { 
+      if (visitorData.data.subscribedTo){
+        const initialState = visitorData.data.subscribedTo
+          .map(subscribedTo => {return {mediaId: subscribedTo.subscriptionId, presences: subscribedTo.currentState} as VideoPresences})
+          .filter(presence => presence.presences.length)
+        setVideoPresences([...initialState])
```

Comment:
presumably this needs to be spread into a new array?

Reply:
...no. Just passing in the array value now.